internal/bgp/bgpfrr/frr.go

```diff
@@ -0,0 +1,242 @@
+package bgpfrr
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"go.universe.tf/metallb/internal/bgp"
+)
+
+// As the MetalLB controller should handle messages synchronously, there should
+// no need to lock this data structure. TODO: confirm this.
+type frrState struct {
+	sessions map[string]*Session
+}
+
+var state *frrState
+
+func init() {
+	state = &frrState{
+		sessions: map[string]*Session{},
+	}
+}
```

Comment:
Yeah, I did experiment with something like this initially. It introduced more churn than I wanted but as the design has progressed I think something like this would be useful.

Reply:
I continued my prototype and I think it looks better. As a result, I pushed a commit that implements a SessionManager. I think this addresses @johananl comments